### PR TITLE
add build and release pipeline

### PR DESCRIPTION
**Pull request by wanglinsong**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                echo "current version: ${REPO_CURRENT_VERSION}"
+
+                sh '''
+                    git config --global --add safe.directory ${PWD}
+                    git config --global user.email "oss-release-bot@prestodb.io"
+                    git config --global user.name "oss-release-bot"
+                '''
+
+                sh '''#!/bin/bash -ex
+                    export GPG_TTY=$TTY
```
Comment (dnskr) [inline Jenkinsfile:60]: It should be the following command instead:
```sh
export GPG_TTY=$(tty)
```
Original command works on Mac, but not on Linux, so in Jenkins logs I found that `GPG_TTY` value is empty:
```sh
[2024-06-12T09:28:00.089Z] + export GPG_TTY=
[2024-06-12T09:28:00.089Z] + GPG_TTY=
[2024-06-12T09:28:00.089Z] + gpg --batch --import ****
```
Test on Mac - both work:
```sh
$ export TEST=$TTY
$ echo $TEST
/dev/ttys002

$ export TEST=$(tty)
$ echo $TEST        
/dev/ttys002
```

Test on Ubuntu - only suggested command works:
```sh
$ export TEST=$TTY
$ echo $TEST
<-(empty line here)

$ export TEST=$(tty)
$ echo $TEST        
/dev/pts/0
```

**Review by dnskr — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (wanglinsong) [reply Jenkinsfile:60]: It is the old maven-gpg-plugin, the version is inherited from parent pom.
I have made a fix.

https://ci.ahana.dev/blue/organizations/jenkins/oss-presto-release%2Fpresto-maven-plugin/detail/PR-15/11/pipeline

**Comment by wanglinsong** [comment]
@dnskr the pipeline should work now. You can try it after this PR is merged.

**Review by dnskr — APPROVED** [review]
Looks good to me

**Pull request merged**